templates/typegen/types.go.tmpl

```diff
@@ -73,7 +77,12 @@ func (x *{{ $typeName }}) UnmarshalJSON(b []byte) error {
         }
 
         for _, m := range rawMessage{{ .Name }} {
+          {{- if contains "." $field.TypeName }}
+          {{-   $m := split "." $field.TypeName }}
+          xxx, err := {{ $m._0 }}.Unmarshal{{ $m._1 }}Interface(*m)
```

Comment:
What is `$m._0` doing here?

Reply:
Template language for `m[0]`.  This resolves referencing interface unmarshallers defined in another package.

Previously generated: `xxx, err := Unmarhsalpackage.TypeInterface(*m)`
Now: `xxx, err := package.UnmarshalTypeInterface(*m)`